src/sentry/api/endpoints/release_threshold_details.py

```diff
@@ -0,0 +1,48 @@
+from django.http import HttpResponse
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.serializers import serialize
+from sentry.api.serializers.rest_framework.project import ProjectField
+from sentry.models import Project
+from sentry.models.release_threshold.releasethreshold import ReleaseThreshold
+
+
+class ReleaseThresholdDetailsGETSerializer(serializers.Serializer):
+    threshold_id = serializers.CharField()
+    project = ProjectField()
+
+
+@region_silo_endpoint
+class ReleaseThresholdDetailsEndpoint(ProjectEndpoint):
+    owner: ApiOwner = ApiOwner.ENTERPRISE
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+
+    def get(self, request: Request, project: Project, threshold_id: str) -> HttpResponse:
+        request.data["project"] = project.slug
+        request.data["threshold_id"] = threshold_id
```

Comment:
instead of doing this, you should write a `convert_args` function that fetches the threshold for you. then you can have your function signature be like

```python
def get(self, request: Request, project: Project, threshold: ReleaseThreshold):
```

i believe then the args will already be set in `request.data` for you. also, since this inherits from `ProjectEndpoint` there is already a `convert_args` function in the parent class that fetches the project for you

i'm also not sure why you need a serializer if you require the path params and you're checking the same path params in the serializer?

Reply:
pretty cool, I didn't know this is how we were fetching the model automatically into the function signature. I found a clear example of it with team SCIM stuff [here](https://github.com/getsentry/sentry/blob/master/src/sentry/scim/endpoints/teams.py#L243-L249)